clusterloader/framework/util.go

```diff
@@ -25,58 +25,59 @@ import (
+func CreateNSIfNotExists(f *framework.Framework, namespaceName string) (*v1.Namespace, error) {
+	var ns *v1.Namespace
+	var err error
+	fullNamespace := getNamespace(f, namespaceName)
 
-// CreatePods creates pods in user defined namspaces with user configurable tuning sets
-func CreatePods(f *framework.Framework, appName string, ns string, labels map[string]string, spec v1.PodSpec, maxCount int, tuning *TuningSetType) {
-	for i := 0; i < maxCount; i++ {
-		framework.Logf("%v/%v : Creating pod", i+1, maxCount)
-		// Retry on pod creation failure
-		for retryCount := 0; retryCount < maxRetries; retryCount++ {
-			_, err := f.ClientSet.Core().Pods(ns).Create(&v1.Pod{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      fmt.Sprintf(appName+"-pod-%v", i),
-					Namespace: ns,
-					Labels:    labels,
-				},
-				Spec: spec,
-			})
-			if err == nil {
-				break
-			}
-			framework.ExpectNoError(err)
+	if fullNamespace == "" {
+		ns, err = f.CreateNamespace(namespaceName, nil)
+		if err != nil {
+			return nil, err
 		}
-		if tuning != nil {
-			// If a rate limit has been defined we wait for N ms between creation
-			if tuning.Pods.RateLimit.Delay != 0 {
-				framework.Logf("Sleeping %d ms between podcreation.", tuning.Pods.RateLimit.Delay)
-				time.Sleep(tuning.Pods.RateLimit.Delay * time.Millisecond)
-			}
-			// If a stepping tuningset has been defined in the config, we wait for the step of pods to be created, and pause
-			if tuning.Pods.Stepping.StepSize != 0 && (i+1)%tuning.Pods.Stepping.StepSize == 0 {
-				verifyRunning := f.NewClusterVerification(
-					&v1.Namespace{
-						ObjectMeta: metav1.ObjectMeta{
-							Name: ns,
-						},
-						Status: v1.NamespaceStatus{},
-					},
-					framework.PodStateVerification{
-						Selectors:   labels,
-						ValidPhases: []v1.PodPhase{v1.PodRunning},
-					},
-				)
+		framework.Logf("Created new namespace: %s", namespaceName)
+	} else {
+		ns, err = f.ClientSet.CoreV1().Namespaces().Get(fullNamespace, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+		framework.Logf("Namespace exists %s ", namespaceName)
+	}
+	return ns, err
+}
 
-				pods, err := verifyRunning.WaitFor(i+1, tuning.Pods.Stepping.Timeout*time.Second)
-				if err != nil {
-					framework.Failf("Error in wait... %v", err)
-				} else if len(pods) < i+1 {
-					framework.Failf("Only got %v out of %v", len(pods), i+1)
-				}
+// getNamespace takes the basename from the config and returns the full generated namespace name
+func getNamespace(f *framework.Framework, baseName string) string {
+	existingNamespaces, _ := f.ClientSet.Core().Namespaces().List(metav1.ListOptions{})
+	for _, value := range existingNamespaces.Items {
+		if value.GenerateName == fmt.Sprintf("e2e-tests-%v-", baseName) {
+			return value.Name
+		}
+	}
+	return ""
+}
+
+// Sleep acts on a TuningSetObject and will sleep for the specified duration
```

Comment:
These functions should be moved to framework/tunning_set.go (or sth like that) - these aren't test-specific.